dotnet/src/webdriver/CookieJar.cs

```diff
@@ -21,38 +21,72 @@
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
 
+#nullable enable
+
 namespace OpenQA.Selenium
 {
     /// <summary>
     /// Defines an interface allowing the user to manipulate cookies on the current page.
     /// </summary>
-    internal class CookieJar : ICookieJar
+    internal sealed class CookieJar : ICookieJar
     {
-        private WebDriver driver;
+        private readonly WebDriver driver;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="CookieJar"/> class.
         /// </summary>
         /// <param name="driver">The driver that is currently in use</param>
+        /// <exception cref="ArgumentNullException">If <paramref name="driver"/> is <see langword="null"/>.</exception>
         public CookieJar(WebDriver driver)
         {
-            this.driver = driver;
+            this.driver = driver ?? throw new ArgumentNullException(nameof(driver));
         }
 
         /// <summary>
         /// Gets all cookies defined for the current page.
         /// </summary>
         public ReadOnlyCollection<Cookie> AllCookies
         {
-            get { return this.GetAllCookies(); }
+            get
+            {
+                object returned = this.driver.InternalExecute(DriverCommand.GetAllCookies, new Dictionary<string, object>()).Value;
+
+                try
+                {
+                    List<Cookie> toReturn = new List<Cookie>();
+                    if (returned is object?[] cookies)
+                    {
+                        foreach (object? rawCookie in cookies)
+                        {
+                            if (rawCookie != null)
+                            {
+                                Cookie newCookie = Cookie.FromDictionary((Dictionary<string, object?>)rawCookie);
+                                toReturn.Add(newCookie);
+                            }
+                        }
+                    }
+
+                    return new ReadOnlyCollection<Cookie>(toReturn);
+                }
+                catch (Exception e)
+                {
+                    throw new WebDriverException("Unexpected problem getting cookies", e);
+                }
+            }
         }
 
         /// <summary>
         /// Method for creating a cookie in the browser
         /// </summary>
         /// <param name="cookie"><see cref="Cookie"/> that represents a cookie in the browser</param>
+        /// <exception cref="ArgumentNullException">If <paramref name="cookie"/> is <see langword="null"/>.</exception>
         public void AddCookie(Cookie cookie)
         {
+            if (cookie is null)
```

Comment:
This is safe because the current behavior throws on `null`

![image](https://github.com/user-attachments/assets/c856dd3c-fcb7-414a-8679-9a277cf958c3)

Reply:
Agree.